partiql-ast/src/main/kotlin/org/partiql/ast/Ast.kt

```diff
@@ -506,4 +512,64 @@ public object Ast {
     public fun sort(expr: Expr, order: Order?, nulls: Nulls?): Sort {
         return Sort(expr, order, nulls)
     }
+
+    //
+    // DDL
+    //
+    @JvmStatic
+    public fun createTable(name: IdentifierChain, columns: List<ColumnDefinition>, constraints: List<TableConstraint>, partitionBy: PartitionBy?, tableProperties: List<KeyValue>): CreateTable {
+        return CreateTable(
+            name,
+            columns,
+            constraints,
+            partitionBy,
+            tableProperties
+        )
+    }
+
+    @JvmStatic
+    public fun columnDefinition(name: Identifier, type: DataType, isOptional: Boolean, constraints: List<AttributeConstraint>, comment: String?): ColumnDefinition {
+        return ColumnDefinition(name, type, isOptional, constraints, comment)
+    }
+
+    @JvmStatic
+    public fun tableConstraintUnique(name: String?, columns: List<Identifier>): TableConstraint.Unique {
+        return TableConstraint.Unique(name, columns)
+    }
+
+    @JvmStatic
+    public fun tableConstraintPrimaryKey(name: String?, columns: List<Identifier>): TableConstraint.PrimaryKey {
+        return TableConstraint.PrimaryKey(name, columns)
+    }
+
+    @JvmStatic
+    public fun tableConstraintCheck(name: String?, searchCondition: Expr): TableConstraint.Check {
+        return TableConstraint.Check(name, searchCondition)
+    }
+
+    @JvmStatic
+    public fun columnConstraintNullable(name: String?, isNullable: Boolean): AttributeConstraint.Null {
+        return AttributeConstraint.Null(name, isNullable)
+    }
+
+    @JvmStatic
+    public fun columnConstraintUnique(name: String?, isPrimaryKey: Boolean): AttributeConstraint.Unique {
+        return AttributeConstraint.Unique(name, isPrimaryKey)
+    }
+
+    @JvmStatic
+    public fun columnConstraintCheck(name: String?, searchCondition: Expr): AttributeConstraint.Check {
+        return AttributeConstraint.Check(name, searchCondition)
+    }
```

Comment:
I haven't checked too closely, but we need to confirm which arguments should be nullable vs non-null in the factory methods compared with the constructor.

Some of the args like `createTable`'s `columns` and `constraints` are nullable in the constructor but marked as non-null in the factory method.